mlrun/k8s_utils.py

```diff
@@ -329,13 +331,38 @@ def get_project_vault_secret_name(
 
         return service_account.secrets[0].name
 
-    def get_project_secret_name(self, project):
+    def get_project_secret_name(self, project) -> str:
         return mlconfig.secret_stores.kubernetes.project_secret_name.format(
             project=project
         )
 
+    def get_auth_secret_name(self, username: str, access_key: str) -> str:
+        sanitized_username = sanitize_username(username)
+        hashed_access_key = self._hash_access_key(access_key)
+        return mlconfig.secret_stores.kubernetes.auth_secret_name.format(
+            sanitized_username=sanitized_username, hashed_access_key=hashed_access_key
+        )
+
+    @staticmethod
+    def _hash_access_key(access_key: str):
+        return hashlib.sha256(access_key.encode()).hexdigest()
+
     def store_project_secrets(self, project, secrets, namespace=""):
         secret_name = self.get_project_secret_name(project)
+        self.store_secrets(secret_name, secrets, namespace)
+
+    def store_auth_secret(self, username: str, access_key: str, namespace="") -> str:
+        secret_name = self.get_auth_secret_name(username, access_key)
+        secret_data = {
+            mlrun.api.schemas.AuthSecretData.get_field_secret_key("username"): username,
+            mlrun.api.schemas.AuthSecretData.get_field_secret_key(
+                "access_key"
+            ): access_key,
+        }
+        self.store_secrets(secret_name, secret_data, namespace, type_="v3io/fuse")
```

Comment:
`v3io/fuse` is a v3io specific string - I think it may make sense to put it as a constant somewhere, or even a config value. Not that I foresee it being changed in the future, but still.

Reply:
It's a string that is representing a k8s secret type, therefore I don't see a better place for it rather than the k8s utils, moved to a constant